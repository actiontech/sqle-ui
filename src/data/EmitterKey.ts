enum EmitterKey {
  Refresh_User_list = 'REFRESH_USER_LIST',
  Refresh_Role_list = 'REFRESH_ROLE_LIST',
  Refresh_Whitelist_List = 'REFRESH_WHITELIST_LIST',

  Refresh_Rule_Template_List = 'REFRESH_RULE_TEMPLATE_LIST',

  Reset_Create_Order_Form = 'RESET_CREATE_ORDER_FORM',

  Reset_Test_Data_Source_Connect = 'RESET_TEST_DATA_SOURCE_CONNECT',

  Reset_Workflow_Template_Form = 'RESET_WORKFLOW_TEMPLATE_FORM',

  Rest_Audit_Plan_Form = 'REST_AUDIT_PLAN_FORM',

  Refresh_Audit_Plan_Record = 'REFRESH_AUDIT_PLAN_RECORD',
}

export default EmitterKey;
