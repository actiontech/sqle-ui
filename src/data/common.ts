export enum ResponseCode {
  SUCCESS = 0,
}

export enum SystemRole {
  admin = 'admin',
}

export enum ModalSize {
  big = 1000,
  mid = 800,
}

export const RuleListDefaultTabKey = 'ALL';
export const instanceListDefaultKey = 'all';
export const ruleTemplateListDefaultKey = 'all';

export const ModalFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 7 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 13 },
  },
};

export const PageFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 8 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 12 },
    md: { span: 10 },
  },
};

export const PageBigFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 6 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 14 },
    // md: { span: 1 },
  },
};

export const FilterFormLayout = {
  labelCol: { span: 7 },
  wrapperCol: { span: 17 },
};

export const FilterFormColLayout = {
  xs: 24,
  sm: 12,
  xl: 8,
  xxl: 6,
};

export const filterFormButtonLayoutFactory = (
  smOffset = 0,
  xlOffset = 0,
  xxlOffset = 0
) => ({
  xs: 24,
  sm: {
    span: 12,
    offset: smOffset,
  },
  xl: {
    span: 8,
    offset: xlOffset,
  },
  xxl: {
    span: 6,
    offset: xxlOffset,
  },
});

export const FilterFormRowLayout = {
  gutter: 24,
};

export enum LoginTypeEnum{
  'ldap'='ldap',
  'sqle'='sqle',
}