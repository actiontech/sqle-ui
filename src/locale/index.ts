import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import StorageKey from '../data/StorageKey';
import LocalStorageWrapper from '../utils/LocalStorageWrapper';
import enUS from './en-US';
import zhCn from './zh-CN';

enum SupportLanguage {
  zhCN = 'zh-CN',
  enUS = 'en-US',
}

i18n.use(initReactI18next).init({
  resources: {
    [SupportLanguage.zhCN]: zhCn,
    [SupportLanguage.enUS]: enUS,
  },
  lng: LocalStorageWrapper.getOrDefault(
    StorageKey.Language,
    SupportLanguage.zhCN
  ),
  fallbackLng: SupportLanguage.zhCN,
  interpolation: {
    escapeValue: false,
  },
});

export { SupportLanguage };
export default i18n;
