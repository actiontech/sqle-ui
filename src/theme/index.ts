import { makeStyles } from '@material-ui/styles';
import { Theme } from '../types/theme.type';
import darkTheme from './dark';
import lightTheme from './light';

enum SupportTheme {
  DARK = 'dark',
  LIGHT = 'light',
}

const ThemeData = {
  [SupportTheme.DARK]: darkTheme,
  [SupportTheme.LIGHT]: lightTheme,
};

const useStyles = makeStyles<Theme>((theme) => ({
  headerBg: {
    backgroundColor: theme.header.background,
    color: theme.header.color,
  },
  editor: {
    border: theme.editor.border,
    paddingRight: 1,
  },
}));

export { SupportTheme, ThemeData };
export default useStyles;
