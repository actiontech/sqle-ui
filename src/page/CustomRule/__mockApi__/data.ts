import { ICustomRuleResV1 } from '../../../api/common';
import { CustomRuleResV1LevelEnum } from '../../../api/common.enum';

export const customRules: ICustomRuleResV1[] = [
  {
    db_type: 'mysql',
    annotation: 'desc desc desc',
    level: CustomRuleResV1LevelEnum.notice,
    rule_id: '1',
    desc: 'name1',
    rule_script: 'w+',
    type: 'rule_type1',
  },
  {
    db_type: 'oracle',
    annotation: 'desc desc desc',
    level: CustomRuleResV1LevelEnum.error,
    rule_id: '2',
    desc: 'name2',
    rule_script: 'w+',
    type: 'rule_type2',
  },
];
