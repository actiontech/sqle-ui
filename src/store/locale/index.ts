import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import StorageKey from '../../data/StorageKey';
import { SupportLanguage } from '../../locale';
import LocalStorageWrapper from '../../utils/LocalStorageWrapper';

type LocaleReduxState = {
  language: string;
};

const initialState: LocaleReduxState = {
  language: LocalStorageWrapper.getOrDefault(
    StorageKey.Language,
    SupportLanguage.zhCN
  ),
};

const locale = createSlice({
  name: 'locale',
  initialState,
  reducers: {
    updateLanguage: (
      state,
      { payload: { language } }: PayloadAction<{ language: SupportLanguage }>
    ) => {
      state.language = language;
      LocalStorageWrapper.set(StorageKey.Language, language);
    },
  },
});

export const { updateLanguage } = locale.actions;

export default locale.reducer;
