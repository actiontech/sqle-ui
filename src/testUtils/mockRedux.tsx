import { Dictionary } from '../types/common.type';
import * as redux from 'react-redux';
import locale from '../store/locale';
import user from '../store/user';
import userManage from '../store/userManage';
import whitelist from '../store/whitelist';
import nav from '../store/nav';
import { combineReducers, createStore } from 'redux';
import React from 'react';
import { Provider } from 'react-redux';

const reducers = combineReducers({
  locale,
  user,
  userManage,
  whitelist,
  nav,
});

export const storeFactory = (initStore: Dictionary = {}) => {
  return createStore(reducers, initStore);
};

export const mockUseSelector = (store?: any) => {
  const spy = jest.spyOn(redux, 'useSelector');
  spy.mockImplementation((getFn: Function) => {
    return getFn(store);
  });
  return spy;
};

export const mockUseDispatch = () => {
  const scopeDispatch = jest.fn();
  const spy = jest.spyOn(redux, 'useDispatch');
  spy.mockImplementation(() => scopeDispatch);
  return { spy, scopeDispatch };
};

export const mockUseLocation = () => {}


export const CustomProvider: React.FC<{ initStore: Dictionary }> = (props) => {
  return (
    <Provider store={storeFactory(props.initStore)}>{props.children}</Provider>
  );
};
