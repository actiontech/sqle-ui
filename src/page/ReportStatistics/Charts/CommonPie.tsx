import { Pie } from '@ant-design/plots';
import { useTheme } from '@material-ui/styles';
import { useMemo } from 'react';
import { useSelector } from 'react-redux';
import { CommonChartsColors, CommonPieProps } from '.';
import { IReduxState } from '../../../store';
import { Theme } from '../../../types/theme.type';
import { floatToPercent } from '../../../utils/Math';
import reportStatisticsData from '../index.data';

const { rowHeight } = reportStatisticsData;
const CommonLine: React.FC<CommonPieProps> = (props) => {
  const {
    h,
    padding = 'auto',
    color = CommonChartsColors,
    radius = 1,
    label = {
      type: 'inner',
      offset: '-30%',
      content: ({ percent }) => `${floatToPercent(percent)}%`,
      style: {
        fontSize: 14,
        textAlign: 'center',
      },
    },
    interactions = [
      {
        type: 'element-active',
      },
    ],
  } = props;
  const currentTheme = useSelector<IReduxState, string>(
    (state) => state.user.theme
  );
  const language = useSelector<IReduxState, string>(
    (state) => state.locale.language
  );
  const theme = useTheme<Theme>();
  const height = useMemo(() => {
    return rowHeight * h + theme.common.padding * (h - 1) - 80;
  }, [h, theme.common.padding]);
  return (
    <Pie
      {...props}
      padding={padding}
      radius={radius}
      label={label}
      interactions={interactions}
      theme={currentTheme}
      locale={language}
      height={height}
      color={color}
    />
  );
};

export default CommonLine;
