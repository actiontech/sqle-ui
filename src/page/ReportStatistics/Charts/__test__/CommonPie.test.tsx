import { PieConfig } from '@ant-design/plots';
import { useTheme } from '@material-ui/styles';
import { SupportLanguage } from '../../../../locale';
import { mockUseSelector } from '../../../../testUtils/mockRedux';
import { SupportTheme } from '../../../../theme';
import CommonPie from '../CommonPie';
import { pieData } from './index.data';
import { render } from '@testing-library/react';

const config: PieConfig = {
  data: pieData,
  angleField: 'percent',
  colorField: 'instance_type',
};

jest.mock('@material-ui/styles', () => {
  return {
    ...jest.requireActual('@material-ui/styles'),
    useTheme: jest.fn(),
  };
});

describe('test CommonLine', () => {
  const useThemeMock: jest.Mock = useTheme as jest.Mock;

  beforeEach(() => {
    mockUseSelector({
      user: { theme: SupportTheme.LIGHT },
      locale: { language: SupportLanguage.zhCN },
    });
    useThemeMock.mockReturnValue({ common: { padding: 24 } });
  });

  test('should match snapshot', () => {
    const { container } = render(<CommonPie h={3} {...config} />);
    expect(container).toMatchSnapshot();
  });
});
