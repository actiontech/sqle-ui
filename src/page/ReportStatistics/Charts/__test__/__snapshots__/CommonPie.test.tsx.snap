// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CommonLine should match snapshot 1`] = `
<div>
  <div
    data-chart-source-type="G2Plot"
    size-sensor-id="1"
    style="height: inherit;"
  >
    <div
      style="position:relative;"
    >
      <canvas
        height="328"
        style="width: 400px; height: 328px; display: inline-block; vertical-align: middle;"
        width="400"
      />
    </div>
  </div>
</div>
`;
